feat: convert package to ESM

hereje opened the pull request:
test files:
- add vitest
- remove jest
- add vitest api globally
- update test files
- replace jest config with vitest config

Fixes: #20

hereje committed:
feat: replace jest with vitest

- add vitest
- remove jest
- add vitest api globally
- update test files
- replace jest config with vitest config

malinovic commented:
Hello @hereje ,
First of all thank you for your effort. I will have a look into it and provide you with feedback.
Is this PR still in work? It is still marked as draft.

hereje committed:
fix: exclude vitest config file from rootDir

- update tsconfig.json config so that vitest config file does not
mess up wwith the current data type generation
